packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSourceBase.ts

```diff
@@ -324,6 +322,8 @@ export const useGridDataSourceBase = <Api extends GridPrivateApiPro>(
     [apiRef],
   );
 
+  const debouncedFetchRows = React.useMemo(() => debounce(fetchRows, 0), [fetchRows]);
```

Comment:
I wonder if using a microtask callback wouldn't have been better than a timeout (aka task). It would keep the overall update to within the same event loop cycle.